File: apps/Applications/ios/Local For Vocal/Local For Vocal/Sources/Views/Components/ProductCardView.swift
import SwiftUI

struct ProductCardView: View {
    let product: Product
    let width: CGFloat
    var onAdd: (() -> Void)?
    
    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            // Image Container
            ZStack(alignment: .topTrailing) {
                Color(hex: "#F3F4F6") // Placeholder background
                
                if let imageUrl = product.images.first, let url = URL(string: imageUrl) {
                    AsyncImage(url: url) { image in
                        image
                            .resizable()
                            .aspectRatio(contentMode: .fill)
                    } placeholder: {
                        ProgressView()
                    }
                    .frame(width: width, height: 160)
                    .clipped()
                } else {
                    Image(systemName: "photo")
                        .font(.system(size: 40))
                        .foregroundColor(.gray)
                        .frame(width: width, height: 160)
                }
                
                // Wishlist Button
                Button(action: {
                    // Wishlist action
                }) {
                    Circle()
                        .fill(Color.white)
                        .frame(width: 32, height: 32)
                        .shadow(color: .black.opacity(0.1), radius: 2, x: 0, y: 1)
                        .overlay(
                            Image(systemName: "heart")
                                .font(.system(size: 16))
                                .foregroundColor(Color(hex: "#4F46E5"))
                        )
                }
                .padding(8)
                
                // Discount Badge
                if let discount = product.discountPercentage, discount > 0 {
                    Text("-\(discount)%")
                        .font(.system(size: 10, weight: .bold))
                        .foregroundColor(.white)
                        .padding(.horizontal, 6)
                        .padding(.vertical, 2)
                        .background(Color(hex: "#EF4444"))
                        .cornerRadius(4)
                        .padding(8)
                        .frame(maxWidth: .infinity, alignment: .topLeading)
                }
            }
            .frame(height: 160)
            
            // Content
            VStack(alignment: .leading, spacing: 4) {
                Text(product.category.uppercased())
                    .font(.system(size: 10, weight: .semibold))
                    .foregroundColor(Color(hex: "#6B7280"))
                
                Text(product.name)
                    .font(.system(size: 14, weight: .semibold))
                    .foregroundColor(Color(hex: "#111827"))
                    .lineLimit(2)
                    .frame(height: 40, alignment: .topLeading)
                
                HStack(alignment: .center) {
                    VStack(alignment: .leading, spacing: 0) {
                        Text("₹\(Int(product.price))")
                            .font(.system(size: 16, weight: .bold))
                            .foregroundColor(Color(hex: "#4F46E5"))
                        
                        if let mrp = product.mrp, mrp > product.price {
                            Text("₹\(Int(mrp))")
                                .font(.system(size: 10))
                                .foregroundColor(Color(hex: "#9CA3AF"))
                                .strikethrough()
                        }
                    }
                    
                    Spacer()
                    
                    // Rating
                    if let rating = product.rating {
                        HStack(spacing: 2) {
                            Image(systemName: "star.fill")
                                .font(.system(size: 10))
                                .foregroundColor(Color(hex: "#F59E0B"))
                            Text(String(format: "%.1f", rating))
                                .font(.system(size: 12, weight: .semibold))
                                .foregroundColor(Color(hex: "#B45309"))
                        }
                        .padding(.horizontal, 6)
                        .padding(.vertical, 2)
                        .background(Color(hex: "#FFFBEB"))
                        .cornerRadius(4)
                    }
                }
                .padding(.top, 4)
            }
            .padding(12)
        }
        .frame(width: width)
        .background(Color.white)
        .cornerRadius(16)
        .shadow(color: .black.opacity(0.05), radius: 4, x: 0, y: 2)
    }
}

File: apps/Applications/android/app/src/main/java/com/localforvocalstartup/app/ui/components/ProductCardView.kt
package com.localforvocalstartup.app.ui.components
// Cache buster 1

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.FavoriteBorder
import androidx.compose.material.icons.filled.Star
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextDecoration
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import coil.compose.AsyncImage
import com.localforvocalstartup.app.data.model.Product

@Composable
fun ProductCardView(
    product: Product,
    modifier: Modifier = Modifier,
    onClick: () -> Unit = {}
) {
    Card(
        modifier = modifier
            .shadow(2.dp, RoundedCornerShape(12.dp))
            .clickable { onClick() },
        shape = RoundedCornerShape(12.dp),
        colors = CardDefaults.cardColors(containerColor = Color.White)
    ) {
        Column {
            // Image Container
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(160.dp)
                    .background(Color(0xFFF3F4F6))
            ) {
                AsyncImage(
                    model = product.images.firstOrNull(),
                    contentDescription = null,
                    modifier = Modifier.fillMaxSize(),
                    contentScale = ContentScale.Crop
                )

                // Wishlist Button (Top Right)
                IconButton(
                    onClick = { /* Wishlist */ },
                    modifier = Modifier
                        .align(Alignment.TopEnd)
                        .padding(8.dp)
                        .size(32.dp)
                        .background(Color.White, CircleShape)
                        .shadow(2.dp, CircleShape)
                ) {
                    Icon(
                        imageVector = Icons.Default.FavoriteBorder,
                        contentDescription = "Wishlist",
                        tint = Color(0xFF4F46E5),
                        modifier = Modifier.size(18.dp)
                    )
                }

                // Discount Badge (Top Left)
                if ((product.discountPercentage ?: 0) > 0) {
                    Text(
                        text = "-${product.discountPercentage}%",
                        fontSize = 10.sp,
                        fontWeight = FontWeight.Bold,
                        color = Color.White,
                        modifier = Modifier
                            .align(Alignment.TopStart)
                            .padding(8.dp)
                            .background(Color(0xFFEF4444), RoundedCornerShape(4.dp))
                            .padding(horizontal = 6.dp, vertical = 2.dp)
                    )
                }
            }

            // Content
            Column(modifier = Modifier.padding(12.dp)) {
                // Category
                if (!product.category.isNullOrEmpty()) {
                    Text(
                        text = product.category.uppercase(),
                        fontSize = 10.sp,
                        fontWeight = FontWeight.SemiBold,
                        color = Color(0xFF6B7280)
                    )
                }

                // Name
                Text(
                    text = product.displayName,
                    fontSize = 14.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = Color(0xFF111827),
                    maxLines = 2,
                    overflow = TextOverflow.Ellipsis,
                    modifier = Modifier.height(40.dp)
                )
                
                Spacer(modifier = Modifier.height(4.dp))

                Row(
                    modifier = Modifier.fillMaxWidth(),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    // Prices
                    Column {
                        Text(
                            text = "₹${product.price.toInt()}",
                            fontSize = 16.sp,
                            fontWeight = FontWeight.Bold,
                            color = Color(0xFF4F46E5)
                        )
                        if ((product.mrp ?: 0.0) > product.price) {
                            Text(
                                text = "₹${product.mrp?.toInt()}",
                                fontSize = 10.sp,
                                textDecoration = TextDecoration.LineThrough,
                                color = Color(0xFF9CA3AF)
                            )
                        }
                    }

                    Spacer(modifier = Modifier.weight(1f))

                    // Rating Pill
                    if (product.rating != null) {
                        Row(
                            verticalAlignment = Alignment.CenterVertically,
                            modifier = Modifier
                                .background(Color(0xFFFFFBEB), RoundedCornerShape(4.dp))
                                .padding(horizontal = 6.dp, vertical = 2.dp)
                        ) {
                            Icon(
                                imageVector = Icons.Default.Star,
                                contentDescription = null,
                                tint = Color(0xFFF59E0B),
                                modifier = Modifier.size(10.dp)
                            )
                            Spacer(modifier = Modifier.width(2.dp))
                            Text(
                                text = String.format("%.1f", product.rating),
                                fontSize = 12.sp,
                                fontWeight = FontWeight.SemiBold,
                                color = Color(0xFFB45309) // Amber 800
                            )
                        }
                    }
                }
            }
        }
    }
}
